2023-Oct-18/Dave/word_grid.py

```diff
@@ -0,0 +1,76 @@
+import random
+import string
+from point import Point
+from direction import Direction
+
+
+class WordGrid:
+    def __init__(self, rows: int, cols: int):
+        self.rows = rows
+        self.cols = cols
+        self.initialise_word_grid()
+
+    def __repr__(self):
+        return self.get_stringified_word_grid()
+
+    def __iter__(self):
+        for row in self.word_grid:
+            yield row
+
+    def initialise_word_grid(self):
+        self.word_grid = [[" " for col in range(
+            self.cols)] for row in range(self.rows)]
+
+    def get_stringified_word_grid(self) -> str:
+        output = ""
+        for row in self.word_grid:
+            for char in row:
+                output += char + " "
+            output += "\n"
+        return output
```

Comment:
Cursory research suggests that join() is faster than iterating with +=, so the 'for char in row' block is an easy (and agreeable) fix.

However, just for clarity, the use of += to concatenate two strings is still fine outside of an iterator?

Reply:
Both are fine just a lack of consistency looks odd (plus if I don't know what join and += do you have to google both)

Reply:
If I try to do this, though:

`"first".join("second")`

I get:

`sfirstefirstcfirstofirstnfirstd`

So it's not clear to me presently how to concatenate two strings with `join()`. Python documentation seems to suggest that the above is correct and the inconsistency is because they're doing different things.

Let me know if you know better.